src/main/java/uk/gov/digital/ho/hocs/queue/migration/MigrationCaseData.java

```diff
@@ -0,0 +1,106 @@
+
+    private final ReadContext ctx;
+    private final String jsonBody;
+
+    public MigrationCaseData(String jsonBody) {
+        this.jsonBody = jsonBody;
+        ctx = JsonPath.parse(jsonBody);
+    }
+
+    @Override
+    public LocalDate getDateReceived() {
+        return LocalDate.parse(ctx.read(CREATION_DATE));
+    }
+
+    @Override
+    public String getComplaintType() {
+        return ctx.read(COMPLAINT_TYPE);
+    }
+
+    @Override
+    public List<ComplaintCorrespondent> getComplaintCorrespondent() {
+
+        List<ComplaintCorrespondent> correspondents = new ArrayList<>();
+
+        try {
+            String applicantType = ctx.read(APPLICANT_TYPE);
+            if (applicantType.equals("APPLICANT")) {
+                ComplaintCorrespondent applicantCorrespondent = new ComplaintCorrespondent(ctx.read(APPLICANT_APPLICANT_NAME), CorrespondentType.COMPLAINANT);
+                optionalString(ctx, APPLICANT_APPLICANT_EMAIL).ifPresent(applicantCorrespondent::setEmail);
+                optionalString(ctx, APPLICANT_APPLICANT_PHONE).ifPresent(applicantCorrespondent::setTelephone);
+                correspondents.add(applicantCorrespondent);
+            } else if (applicantType.equals("AGENT")) {
+                ComplaintCorrespondent applicantCorrespondent = new ComplaintCorrespondent(ctx.read(AGENT_APPLICANT_NAME), CorrespondentType.COMPLAINANT);
+                ComplaintCorrespondent agentCorrespondent = new ComplaintCorrespondent(ctx.read(AGENT_AGENT_NAME), CorrespondentType.THIRD_PARTY_REP);
+                optionalString(ctx, AGENT_AGENT_EMAIL).ifPresent(agentCorrespondent::setEmail);
+                optionalString(ctx, AGENT_AGENT_PHONE).ifPresent(agentCorrespondent::setTelephone);
+                //First correspondent added becomes the primary correspondent for the case.
+                correspondents.add(applicantCorrespondent);
+                correspondents.add(agentCorrespondent);
+            } else {
+                throw new IllegalStateException("APPLICANT_TYPE Unknown : " + applicantType);
+            }
+        } catch(PathNotFoundException e){
+            log.info("getComplaintCorrespondent, no correspondents found for case.");
+        }
+
+        return correspondents;
+    }
+
+    @Override
+    public String getFormattedDocument() {
+        String formattedText = jsonBody; // Fall back if conversion fails
+        try {
+            JSONToSimpleTextConverter jsonToSimpleTextConverter = new JSONToSimpleTextConverter(jsonBody);
+            formattedText = jsonToSimpleTextConverter.getConvertedOutput();
+        } catch (IOException e) {
+            log.warn("Document formatting failed due to : {}", e.getMessage());
+        }
+        return formattedText;
+    }
+
+    @Override
+    public String getRawPayload() {
+        return jsonBody;
+    }
+
+    private Optional<String> optionalString(ReadContext ctx, String path) {
```

Comment:
This is also the same as `UKVIComplaintData`.

Reply:
Fixed as suggested, extracted into an abstract class and inherited.